### PR TITLE
Feature/product reviews

### PR DESCRIPTION
**Pull request by daniel-espinoza-79**
I've implemented enhancements to the product request system to ensure a more secure and fair user experience. Here are the key changes:

Mandatory Authentication: Users are now required to be logged in to submit product reviews. This enhances security and authenticity while safeguarding against fraudulent reviews.

One Review Per Product Limit: We've introduced a system that allows each user to leave only one review per product. This prevents duplicate or excessive reviews and promotes an equitable distribution of user opinions.

Lazy loading of reviews : it has the ability to show more or less reviews per product 

Summary Rating :  it contains a summary of all the reviews of each star 

to test this branch, it has to be done with local database 

db.sneakers.updateMany(
   { reviews: { $exists: true, $type: "object" } },
   {
      $set: {
         "reviews": {
            "RatingSummary": {
               "star1": 0,
               "star2": 0,
               "star3": 0,
               "star4": 0,
               "star5": 0
            },
            "total": 0
         }
      }
   }
)

We are running this scrypt because I changed the database schema, and when the code reviews are done, I will run it on the actual MongoDB Atlas cloud service.

### CODE REVIEW

**Review by luiggy-mamani-jalafund — COMMENTED** [review]
## BUG
Product not found to see the product details and also their reviews, that is caused by a backend error just in your branch.
This is a screenshot about how the backend cannot find the document.

Captura desde 2023-11-13 21-58-52
Updated : The bug is not of this branch

File: backend/product_management/context/controllers/requests/ReviewRequester.go [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+	err = database.SneakerCollection.FindOne(c.Context(), bson.M{"_id": sneakerObjId}).Decode(&sneaker)
+	if err != nil {
+		return c.Status(500).SendString("Error finding sneaker")
+	}
+
+
+	if err != nil {
+		return c.Status(500).SendString("Error inserting review")
+	}
+
```
Comment (VictorCespedesCC) [inline backend/product_management/context/controllers/requests/ReviewRequester.go:53]: Unnecessarily repeated code

File: frontend/src/components/reviews/ProductReview.jsx [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+
+    const showMoreReviews = () => {
+        let index = reviews.length === 0 ? 0 : reviews.length
+        getReviewsByAmount(product._id, index, 3).then((data) => {
+            let newReviews = [...reviews, ...data]
+            setReviews(newReviews)
+        }).catch((err) => {
+            console.log(err)
+        })
```
Comment (VictorCespedesCC) [inline frontend/src/components/reviews/ProductReview.jsx:22]: Currently all errors are printed in console, you can be more specific with the error, to improve future debugging:
.then((data) => {
         let newReviews = [...reviews, ...data];
         setReviews(newReviews);
      })
      .catch((err) => {
         console.error("Error fetching reviews:", err);
         // Manejar el error de manera adecuada (mostrar un mensaje al usuario, etc.).
      });

**Review by VictorCespedesCC — CHANGES_REQUESTED** [review]

File: backend/product_management/context/controllers/requests/ReviewRequester.go (hunk 1 above)
Comment (daniel-espinoza-79) [reply backend/product_management/context/controllers/requests/ReviewRequester.go:53]: The code is not repeated it is for validation and a good response to the client

**Review by VictorCespedesCC — APPROVED** [review]

**Review by luiggy-mamani-jalafund — COMMENTED** [review]
Error decoding key reviews, that causes the details of a sneaker to be inaccessible.

Captura desde 2023-11-14 18-49-03

**Comment by luiggy-mamani-jalafund** [comment]
> Error decoding key reviews, that causes the details of a sneaker to be inaccessible.
> 
> Captura desde 2023-11-14 18-49-03



> Error decoding key reviews, that causes the details of a sneaker to be inaccessible.
> 
> Captura desde 2023-11-14 18-49-03

This error will be corrected when the structure of a sneaker in our database is changed.

**Review by luiggy-mamani-jalafund — APPROVED** [review]

**Pull request merged**